episodes/web-location-identifiers.Rmd

```diff
@@ -0,0 +1,257 @@
+11. To do so on the web, we need the address...
+12. where we can find the resource with a specific name. The address alone is not enough, though. We also need information on how we can retrieve the resource...
+13. the communication protocol. We will focus on this technology in a minute. These information are given as a Uniform Resource Locator, URL: a series of characters, that specify the address of a resource and the protocol necessary to retrieve this resource.
+14. Both, URN and URL, can be used to identify a specific resource. You either need the name or the location, to identify data on the web. So URN and URL **are** Uniform Resource Identifier.
+15. We will further focus on URLs and the technology hidden behind this string of characters, that we enter to open a resource on the web.
+16. And for this we take the URL that retrieves the information for our gene example.
+17. So let's decode this character set: The resource we are looking for, is...
+18. hosted on a specific server. On this server, a certain file path needs to be followed, that will lead to the resource. The final resource might be stored in a data base structure. So the necessary query parameter are specified that lead to the data base entry for the requested resource. And don't forget, we will need to follow a specific protocol, in this case HTTPS, in order to communicate with the host server. HTTPS is an extension of the Hypertext Transfer Protocol, that uses encryption technology for secure communication on the network.
+19. If we now enter this URL in our browser, the browser sends a HTTP request to the host server.
+20. This request specifies, that the browser wants to **GET** a resource stored under a **specific path** from the **host** and follows the specific HTTP protocol that is used for communication, so which is also expected by the browser of the server response.
+21. The host server than looks up the specified file path locally. Finds the requested file and...
+22. sends back the HTTP response, with the requested resource.
+23. The response also includes a status code. The code "200: ok" tells the client server, that send the request, "I have found the file you requested. Here you go!". There are various status codes for server-client communication, but for now, we will focus on the most common ones. The client, your browser, receives the response with the resource attached and..
+24. displays it.
+25. But I'm fairly sure, we have all experienced this situation: You are looking for publications online, be it data or a journal article. The abstract is super promising and you want to retrieve the PDF...
+26. but instead of the publication, you are faced with an **Error 404**. 
```

Comment:
Going further with the coffee order: 

404: "you asked for something I can't find / not on the menu"
or other error codes could be equivalent to "we have no coffee beans left, I can't prepare your coffee now, try again later"